### PR TITLE
Make HTTP middleware optional

### PR DESCRIPTION
**Pull request by MichelHollands**
Add an option to change the Router and an option to not add the default middleware.

Signed-off-by: Michel Hollands <michel.hollands@grafana.com>

### CODE REVIEW

File: server/server.go [hunk 1]
```diff
@@ -276,7 +285,10 @@ func New(cfg Config) (*Server, error) {
 			RouteMatcher: router,
 		},
 	}
-
+	httpMiddleware := []middleware.Interface{}
+	if !cfg.DoNotAddDefaultHTTPMiddleware {
+		httpMiddleware = append(httpMiddleware, defaultHTTPMiddleware...)
```
Comment (bboreham) [inline server/server.go:290]: this `append` seems overkill; a simple assignment would also work?

File: server/server.go [hunk 2]
```diff
@@ -276,7 +285,10 @@ func New(cfg Config) (*Server, error) {
...
+	}
 	httpMiddleware = append(httpMiddleware, cfg.HTTPMiddleware...)
```
Comment (bboreham) [inline server/server.go:292]: or you could put the `DoNotAddDefaultHTTPMiddleware` condition on this line, so you either do an append or take `cfg.HTTPMiddleware` unaltered

Comment (MichelHollands) [reply server/server.go]: That indeed reads better.

**Review by bboreham — APPROVED** [review]
Thanks.

**Pull request merged**